src/components/Person/Person.jsx

```diff
@@ -1 +1,43 @@
 // export const Person = ({ person }) => ();
+import React from 'react';
+
+export const Person = ({ person }) => (
+  <section className="Person">
+    <h2 className="Person__name">
+      {(`My name is ${person.name}`)}
+    </h2>
+
+    {person.age && (
+      <p className="Person__age">
+        {`I am ${person.age}`}
+      </p>
+    )}
+
+    <p className="Person__partner">
+      {(() => {
+        switch (person.isMarried) {
+          case false:
+            return `I am not married`;
+          case true:
+            switch (person.sex) {
+              case 'm':
+                return `${person.partnerName} is my wife`;
+
+              case 'f':
+                return `${person.partnerName} is my husband`;
+
+              default:
+                break;
+            }
+
+            break;
+
+          default:
+            break;
+        }
+
+        return this;
```

Comment:
I like your approach but wouldn't it be easier and shorter like this?

```
1.   const partner = (sex === 'm')
2.     ? 'wife'
3.     : 'husband';
4. 
5.   const status = (isMarried)
6.     ? `${partnerName} is my ${partner}`
7.     : 'I am not married';
```

Basically the same thing but shorter.